chef-config/lib/chef-config/package_task.rb

```diff
@@ -0,0 +1,214 @@
+    end
+
+    def version_rb_path
+      File.expand_path("lib/#{module_path_name}/version.rb", root_path)
+    end
+
+    def version
+      IO.read(File.expand_path(version_file_path, root_path)).strip
+    end
+
+    def full_package_dir
+      File.expand_path(package_dir, root_path)
+    end
+
+    def with_clean_env(&block)
+      if defined?(Bundler)
+        Bundler.with_clean_env(&block)
+      else
+        block.call
+      end
+    end
+
+    def define
+      fail 'Need to provide package root and module name' if root_path.nil? || module_name.nil?
+
+      desc 'Build Gems of component dependencies'
+      task :package_components do
+        component_full_paths.each do |component_path|
+          Dir.chdir(component_path) do
+            sh 'rake package'
+          end
+        end
+      end
+
+      task :package => :package_components
+
+      desc 'Build and install component dependencies'
+      task :install_components => :package_components do
+        component_full_paths.each do |component_path|
+          Dir.chdir(component_path) do
+            sh 'rake install'
+          end
+        end
+      end
+
+      task :install => :install_components
+
+      desc 'Clean up builds of component dependencies'
+      task :clobber_component_packages do
+        component_full_paths.each do |component_path|
+          Dir.chdir(component_path) do
+            sh 'rake clobber_package'
+          end
+        end
+      end
+
+      task :clobber_package => :clobber_component_packages
+
+      desc 'Update the version number for component dependencies'
+      task :update_components_versions do
+        component_full_paths.each do |component_path|
+          Dir.chdir(component_path) do
+            sh 'rake version'
+          end
+        end
+      end
+
+      desc 'Regenerate lib/#{@module_path_name}/version.rb from VERSION file'
+      task :version => :update_components_versions do
+        contents = <<-VERSION_RB
+# Copyright:: Copyright (c) 2010-2015 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+# NOTE: This file is generated by running `rake version` in the top level of
+# this repo. Do not edit this manually. Edit the VERSION file and run the rake
+# task instead.
+#!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+module #{module_name}
```

Comment:
This doesn't work because a long time ago we incorrectly made the top level constant `Chef` a class and it's probably too late to fix it :crying_cat_face: